spec/system/user_logs_in_with_password_spec.rb

```diff
@@ -0,0 +1,49 @@
+require "rails_helper"
+
+RSpec.describe "Authenticating with a password" do
+  def submit_login_form(email, password)
+    fill_in "Email", with: email
+    fill_in "Password", with: password
+    click_button "Log in"
+  end
+
+  let(:password) { "p4assw0rd" }
+  let!(:user) { create(:user, password: password, password_confirmation: password) }
+
+  before do
+    visit sign_up_path
+  end
+
+  context "when logging in with incorrect credentials" do
+    it "displays an error when the email address is wrong" do
+      submit_login_form("wrong@example.com", password)
+      expect(page).to have_text("Invalid Email or password.")
+    end
+
+    it "displays an error when the password is wrong" do
+      submit_login_form(user.email, "wr0ng")
+      expect(page).to have_text("Invalid Email or password.")
+    end
+
+    it "displays a message on the last login attempt" do
+      allow(User).to receive(:maximum_attempts).and_return(2)
+
+      submit_login_form(user.email, "wr0ng")
+      expect(page).to have_text("You have one more attempt before your account is locked.")
+    end
+
+    it "displays a message when the user got locked out" do
+      allow(User).to receive(:maximum_attempts).and_return(1)
+
+      submit_login_form(user.email, "wr0ng")
+      expect(page).to have_text("Your account is locked.")
```

Comment:
This is not necessarily related to your PR, but I was curious to know can we configure this to specify how long your account will be locked for. i.e. "You account is locked. Please try again in an hour". This will reduce the amount of support requests and panic we get from users.

Reply:
It's the option `config.unlock_in` in `config/initializers/devise.rb`. Note that time-based unlocking is actually a convenience feature, normally the account would just be locked until you click the unlock link in the email.